### DISCUSSION TITLE
Transparency for new pixels using rotate

### DISCUSSION THREAD

**Discussion by johndoe46** [libvips/libvips#2639] [Q&A] [answered]
Hello,

I'm using the vips cli to rotate a picture and it works nice and is very fast:

vips rotate test.tif test_vips.tif 12,5 --background "255 255 255"

However, I couldn't find from the function list how to set transparency for new pixels, instead of a color.

Is this possible ?

Thank you.

**Comment by jcupitt** [comment, marked as answer]
Hi @johndoe46,

You need to add an alpha channel before rotating. Something like:

```
vips bandjoin_const test.tif t1.v 255
vips rotate t1.v test_vips.tif 12.5
```

Don't use `background` unless you'd like something like a semi-transparent colour behind the image. Don't use `12,5` -- libvips uses the C convention for number representation, so it's `12.5`.

If you use something like python or ruby instead of the CLI you can do it in one step and gain a bit more speed.

```python
image = pyvips.Image.new_from_file("test.tif")
image = image.bandjoin(255).rotate(12.5)
image.write_to_file("test_vips.tif")
```

You don't need to add an alpha If the image already has one, of course, so this might be better:

```python
image = pyvips.Image.new_from_file(filename)
if not image.hasalpha():
    image = image.bandjoin(255)
image = image.rotate(12.5)
image.write_to_file(new_filename)
```

**Comment by johndoe46** [comment]
Hello John,

Thank you for the detailed and incredibly quick answer !

Indeed, python is what I'll use, so thanks for the example.

About `12.5` vs `12,5`: `12.5` works fine with pyvips. However, using the CLI, `12.5` produces the same output as `12` while `12,5` produces the right output. Maybe my shell is messing with the params, I don't know. Anyway it's fine with pyvips.

Side note (not Vips-specific but since other users may encounter the problem...) on MacOS using homebrew for vips install (8.12.1), one has to change the DYLD_LIBRARY_PATH for pyvips to work:

`export DYLD_LIBRARY_PATH="/opt/homebrew/lib/:$DYLD_LIBRARY_PATH"`

**Reply by jcupitt** [reply]
> About 12.5 vs 12,5: 12.5 works fine with pyvips. However, using the CLI, 12.5 produces the same output as 12 while 12,5 produces the right output.

Oh, that's weird, I'll check.

> Side note (not Vips-specific but since other users may encounter the problem...) on MacOS using homebrew for vips install (8.12.1), one has to change the DYLD_LIBRARY_PATH for pyvips to work:

Yes, you always need to set up your shell if you want to use homebrew software outside homebrew, so you need to set `PKG_CONFIG_PATH`, `PATH`, `DYLD_LIBRARY_PATH`, `GEM_HOME`, etc. etc.
